blocks/priority-list/priority-list.js

```diff
@@ -0,0 +1,105 @@
+import { div, img } from '../../scripts/dom-helpers.js';
+import { CLASS_MAIN_HEADING } from '../../scripts/scripts.js';
+
+const desktopConst = 400;
+const tabConst = 150;
+const tabTopPosition = 100;
+function createCard(card) {
+  const [cardtitle, carddesc, ,] = card.children;
+  cardtitle.textContent = cardtitle.textContent.replace(
+    'world-bank:category/',
+    '',
+  );
+  card.className = cardtitle.textContent;
+  cardtitle.className = 'cardtitle';
+  carddesc.className = 'carddesc';
+}
+
+export default async function decorate(block) {
+  const [subtitle, maintitle, buttontext, buttonlink, ...cards] = [
+    ...block.children,
+  ];
+  subtitle.className = 'subtitle';
+  maintitle.className = CLASS_MAIN_HEADING;
+  const buttonLink = buttonlink.getElementsByTagName('a')[0];
+  buttonLink.innerHTML = buttontext.textContent;
+  buttonLink.title = buttontext.textContent;
+  buttonLink.className = 'button primary';
+  const buttonWrapper = div(
+    { class: 'button-wrapper' },
+    buttonLink.cloneNode(true),
+  );
+  const textContainer = div({ class: 'text-container' });
+  textContainer.append(subtitle);
+  textContainer.append(maintitle);
+  textContainer.append(buttonWrapper);
+  buttonlink.remove();
+  buttontext.remove();
+  const cardsContainer = div({ class: 'cards-container' });
+  cards.forEach((card) => {
+    createCard(card);
+    cardsContainer.append(card);
+  });
+  const leftColumnContainer = div({ class: 'left-column-container' });
+  const rightColumnContainer = div({ class: 'right-column-container' });
+  rightColumnContainer.append(textContainer);
+  rightColumnContainer.append(cardsContainer);
+  const imageContainer = div({ class: 'image-container' });
+  const imgElement = img({ src: cards.at(0).querySelector('img').src });
+  imageContainer.append(imgElement);
+
+  imageContainer.style.backgroundImage = `url(${
+    cards.at(0).querySelector('img').src
+  })`;
+  leftColumnContainer.append(imageContainer);
+  block.append(leftColumnContainer);
+
+  window.addEventListener('scroll', () => {
+    const image = leftColumnContainer.querySelector('.image-container');
```

Comment:
can you please move all this logic to seperate function for better readability. (ex:handleScroll)